src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/MSBuildNuGetProjectSystemFactory.cs

```diff
@@ -31,8 +31,6 @@ internal static class MSBuildNuGetProjectSystemFactory
 
         public async static Task<VsMSBuildProjectSystem> CreateMSBuildNuGetProjectSystemAsync(IVsProjectAdapter vsProjectAdapter, INuGetProjectContext nuGetProjectContext)
         {
-            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
```

Comment:
Only async methods called here need the UI thread, this method itself doesn't have a UI thread affinity.